prog-o-meter.py

```diff
@@ -51,48 +52,54 @@ def __init__(self, _days, _filename, _username):
         self.days = _days
         self.GOAL = 100
         self.rectangle_list = []
+        self.encourage = ["Nice Job!", "The force is strong with you!", "Well Done!", "You are doing it!", "Never, never, never give up.", "If you dream it, you can do it.", "Everything you can imagine is real.", "Hope is a waking dream.", "You are the best!", "Always believe in yourself."]
         self.days_remaining = self.GOAL - self.days
         self.completion_date = self.get_completion_date(self.days_remaining-1)
-        # Tkinter instantiation 
+        self.x = 0
+        # Tkinter instantiation
         self.canvas_layout()
         self.button_layout()
         self.prog_o_meter()
         self.progress()
         self.root.mainloop()
     def canvas_layout(self):
         """Display a Tkinter canvas.
-        
-        Creates a 1200x600 px canvas, with a greeting text, and a text stating how many days the user have left untill they reach the goal.  
-       
+
+        Creates a 1200x600 px canvas, with a greeting text, and a text stating how many days the user have left untill they reach the goal.
+
         Attributes:
             CANVAS_WIDTH: The width of the canvas is hardcoded to 1200 px.
             CANVAS_HEIGHT: The height of the canvas is hardcoded to 600 px.
             canvas: The Tkinter canvas widget.
-            countdown_text: A text element on the canvas, stating how many days the user have left before reaching their goal. 
+            countdown_text: A text element on the canvas, stating how many days the user have left before reaching their goal.
         """
         self.CANVAS_WIDTH = 1200
         self.CANVAS_HEIGHT = 600
         VERTICAL_TEXT_POSITION = 100
         self.canvas = Tk.Canvas(self.root, width = self.CANVAS_WIDTH, height = self.CANVAS_HEIGHT)
         self.canvas.pack()
-        self.canvas.create_text(self.CANVAS_WIDTH/2, VERTICAL_TEXT_POSITION, text = ("".join(("Hello ", self.username))))
+        self.hog = self.canvas.create_text(self.CANVAS_WIDTH/2, VERTICAL_TEXT_POSITION, text = ("".join(("Hello ", self.username))))
         self.countdown_text = self.canvas.create_text(self.CANVAS_WIDTH/2, VERTICAL_TEXT_POSITION+40, justify = Tk.CENTER, text = "".join(("You have ", str(self.days_remaining), " days left!\n\n", "If you code everyday, you will be done with this project on ", self.completion_date)))
+        self.encourage_text = self.canvas.create_text(self.CANVAS_WIDTH/2, VERTICAL_TEXT_POSITION+40, justify = Tk.CENTER, text = "".join(self.encourage[self.new_no(self.x)]))

```

Comment:
I think it would be best to not print an encouragement until after the user clicks the button, so I would suggest we save this until later in the program (I have added a comment where I think this should be moved to). But let me know if you disagree!

Reply:
So, my suggestion would be that we remove this line of code (83), because it prints the encouragement straight away. I think we should wait to encourage the user untill after they click the button :)